packages/toolkit/src/codewhisperer/service/transformByQHandler.ts

```diff
@@ -78,89 +77,123 @@ export async function getOpenProjects() {
 }
 
 /*
- * This function searches for a .class file in the selected project. Then it runs javap on the found .class file to get the JDK version
- * for the project, and sets the version in the state variable. Only JDK8 and JDK11 are supported. It also ensure a pom.xml file is found,
- * since only the Maven build system is supported for now.
+ * This function filters all open projects by first searching for a .java file and then searching for a pom.xml file in all projects.
+ * It also tries to detect the Java version of each project by running "javap" on a .class file of each project.
+ * As long as the project contains a .java file and a pom.xml file, the project is still considered valid for transformation,
+ * and we allow the user to specify the Java version.
  */
-export async function validateProjectSelection(project: vscode.QuickPickItem) {
-    const projectPath = project.description
-    const buildSystem = await checkBuildSystem(projectPath!)
-    if (buildSystem !== BuildSystem.Maven) {
-        void vscode.window.showErrorMessage(CodeWhispererConstants.noPomXmlFoundMessage)
-        telemetry.codeTransform_isDoubleClickedToTriggerInvalidProject.emit({
-            codeTransformSessionId: codeTransformTelemetryState.getSessionId(),
-            codeTransformPreValidationError: 'NonMavenProject',
-            result: MetadataResult.Fail,
-            reason: buildSystem === BuildSystem.Gradle ? buildSystem : 'NoPomFileFound',
-        })
-        throw new ToolkitError('No valid Maven build file found', { code: 'CouldNotFindPomXml' })
+export async function validateOpenProjects(projects: vscode.QuickPickItem[]) {
+    const javaProjects = []
+    for (const project of projects) {
```

Comment:
I think what you're referring to is the code we've had since November (https://github.com/aws/aws-toolkit-vscode/pull/4401/files/66be7c4d834748ff81141fb5caebb59d2c5d11d1#r1482102566) which checks for .class files to be present. I guess I could generalize this `vscode.workspace.findFiles` in a separate method and have it accept the pattern we are searching for (ex. `**/*.java` or `**/*.class`) but that doesn't seem necessary to me.